iceberg/iceberg-rest-server/src/test/java/org/apache/gravitino/iceberg/integration/test/util/IcebergRESTServerManager.java

```diff
@@ -0,0 +1,165 @@
+  }
+
+  public Config getServerConfig() {
+    return serverConfig;
+  }
+
+  public void startIcebergRESTServer() throws Exception {
+    initServerConfig();
+    Optional<Future<?>> future = doStartIcebergRESTServer();
+
+    long beginTime = System.currentTimeMillis();
+    boolean started = false;
+
+    while (System.currentTimeMillis() - beginTime < 1000 * 60) {
+      started = isHttpServerUp(checkUri);
+      if (started || (future.isPresent() && future.get().isDone())) {
+        break;
+      }
+      sleepUninterruptibly(500, TimeUnit.MILLISECONDS);
+    }
+    if (!started) {
+      try {
+        if (future.isPresent()) {
+          future.get().get(1, TimeUnit.SECONDS);
+        }
+      } catch (Exception e) {
+        throw new RuntimeException("IcebergRESTServer start failed", e);
+      }
+      throw new RuntimeException("Can not start IcebergRESTServer");
+    }
+  }
+
+  public void stopIcebergRESTServer() {
+    doStopIcebergRESTServer();
+    sleepUninterruptibly(500, TimeUnit.MILLISECONDS);
+
+    long beginTime = System.currentTimeMillis();
+    boolean started = true;
+    while (System.currentTimeMillis() - beginTime < 3 * 1000 * 60) {
+      sleepUninterruptibly(500, TimeUnit.MILLISECONDS);
+      started = isHttpServerUp(checkUri);
+      if (!started) {
+        break;
+      }
+    }
+    if (started) {
+      throw new RuntimeException("Can not stop IcebergRESTServer");
+    }
+  }
+
+  private void customizeConfigFile(String configTempFileName, String configFileName)
+      throws IOException {
+    Map<String, String> configMap = new HashMap<>();
+    configMap.put(
+        IcebergConfig.ICEBERG_CONFIG_PREFIX + JettyServerConfig.WEBSERVER_HTTP_PORT.getKey(),
+        String.valueOf(RESTUtils.findAvailablePort(2000, 3000)));
+
+    configMap.putAll(customConfigs);
+
+    ITUtils.rewriteConfigFile(configTempFileName, configFileName, configMap);
+  }
+
+  private void initServerConfig() throws IOException {
+    Path configDir = getConfigDir();
+    String gravitinoRootDir = System.getenv("GRAVITINO_ROOT_DIR");
+
+    Path configFile = Paths.get(configDir.toString(), IcebergRESTServer.CONF_FILE);
+    customizeConfigFile(
+        Paths.get(gravitinoRootDir, "conf", IcebergRESTServer.CONF_FILE + ".template").toString(),
+        configFile.toString());
+    this.serverConfig = new ServerConfig();
+    Properties properties = serverConfig.loadPropertiesFromFile(configFile.toFile());
+    serverConfig.loadFromProperties(properties);
+
+    LOG.info("Server config:{}.", serverConfig.getAllConfig());
+
+    JettyServerConfig jettyServerConfig =
+        JettyServerConfig.fromConfig(serverConfig, IcebergConfig.ICEBERG_CONFIG_PREFIX);
+    String host = jettyServerConfig.getHost();
+    int port = jettyServerConfig.getHttpPort();
+    this.checkUri = String.format("http://%s:%d/metrics", host, port);
+    LOG.info("Check uri:{}.", checkUri);
+  }
+
+  private static boolean isHttpServerUp(String testUrl) {
```

Comment:
Can we move this method to module `common`, there are several methods with this name.

Reply:
ok

Reply:
move to `HttpUtils.isHttpServerUp`